### ISSUE TITLE
Aggreagator client unable to parse DDI with no titlStmt/titl inside citation

### ISSUE THREAD

**Issue by matthew-morris-cessda** [cessda/cessda.cdc.versions#547]
These files are not parsed. The failing file is attached: [doi%3A10.11587%2F7LX1BD.zip](https://github.com/cessda/cessda.cdc.versions/files/11460747/doi.3A10.11587.2F7LX1BD.zip).

Traceback from the aggregator:

```text
Traceback (most recent call last):
  File "/root/.local/lib/python3.9/site-packages/
kuha_client/__init__.py", line 677, in upsert_paths
    await self.upsert_from_parser(parser)
  File "/root/.local/lib/python3.9/site-packages/kuha_client/__init__.py", line 659, in upsert_from_parser
    for record in parser.select(methods.collection):
  File "/root/.local/lib/python3.9/site-packages/cdcagg_common/mappings.py", line 178, in studies
    for study in super().studies:
  File "/root/.local/lib/python3.9/site-packages/kuha_common/document_store/mappings/ddi/codebook.py", line 272, in studies
    for mapped_related_publications in self._iter_related_publications_as_mapped_params():
  File "/root/.local/lib/python3.9/site-packages/kuha_common/document_store/mappings/ddi/codebook.py", line 134, in _iter_related_publications_as_mapped_params
    raise MappingError("Invalid DDI. No titlStmt/titl inside citation")
kuha_common.document_store.mappings.exceptions.MappingError: Invalid DDI. No titlStmt/titl inside citation
Configured to continue on parse errors. Bypassing file '/xml_sources/validated/AUSSDA/all_published/oai_ddi/doi%3A10.11587%2F6UXAAG.xml'
Unable to parse file '/xml_sources/validated/AUSSDA/all_published/oai_ddi/doi%3A10.11587%2F7LX1BD.xml'. Is the file valid?
```

@toni-sissala - are these files possible to parse in a sensible way?

**Comment by toni-sissala** [comment]
Yes. The failing element is within a relPubl parent which is used to define related publications. There is no requirement for a study to have related publications, at least not in the aggregator. I see two possible paths to take when encountered this issue (The issue is the missing titl -element).

1. **Do not gather any related publications for the record** This is a straightforward and easy fix. Basically means that instead of raising the MappingError, the method returns nothing. I assume this should have no side effects to other functionality.
2. **Gather related publications, but bypass the title**. This is more complex and requires some planning. The method is already quite complex and I would like to refactor it to decrease complexity before adding more logic. The example file looks like it does not provide any other info to related publications besides the IDNo, which is a DOI. Also this might have side-effects to the OAI-PMH Repo Handler which renders the relPubls to DDI codebook. At least I must make sure it renders the titl-element even if it would have no content, since otherwise the Aggregator would produce invalid DDI.

I propose going with option 1. The correct solution for this issue would be to fix the DDI according to the standard and if I understand correctly, that would be the long term plan for these cases.

Any thoughts?

**Comment by matthew-morris-cessda** [comment]
My perspective is that option 1 is the more pragmatic approach so we can at least get the data in, option 2 could be a separate issue that can be tackled later.

**Comment by MortenSikt** [comment]
This is not a requirement for the catalogues harvesting our endpoint. So we agree that option 1 is the best solution,

**Comment by markusjt** [comment]
I also agree that option 1 is the way to go here even if option 2 would also technically be possible. It's preferable to get things like this fixed at the source (or at least before it goes to the Aggregator).

**Comment by toni-sissala** [comment]
This is implemented and released in cessda.cdc.aggregator.client 0.6.0.
